pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/reader/ForwardIndexReader.java

```diff
@@ -916,4 +920,72 @@ default byte[][] getBytesMV(int docId, T context) {
   default int getNumValuesMV(int docId, T context) {
     throw new UnsupportedOperationException();
   }
+
+  // Functions for recording absolute buffer byte ranges accessed while reading a given docId
+
+  /**
+   * Returns whether the forward index supports recording the byte ranges accessed while reading a given docId
+   */
+  default boolean isBufferByteRangeInfoSupported() {
+    return false;
+  }
+
+  /**
+   * Returns a list of {@link ByteRange} that represents all the distinct
```

Comment:
Let's add some javadoc stating only one set of the APIs (either fixed length based or record based) should be used

Reply:
Ack